x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/assignment/TrainedModelAssignment.java

```diff
@@ -289,13 +292,14 @@ public AdaptiveAllocationsSettings getAdaptiveAllocationsSettings() {
     }
 
     public boolean isSatisfied(Set<String> assignableNodeIds) {
-        int allocations = nodeRoutingTable.entrySet()
+        // assignableNodeIds == all the ML Nodes in the cluster
+        int sumOfCurrentAllocationsOnAllStartingOrStartedNodes = nodeRoutingTable.entrySet()
             .stream()
             .filter(e -> assignableNodeIds.contains(e.getKey()))
             .filter(e -> e.getValue().getState().isAnyOf(RoutingState.STARTING, RoutingState.STARTED))
-            .mapToInt(e -> e.getValue().getTargetAllocations())
+            .mapToInt(e -> e.getValue().getCurrentAllocations())
             .sum();
-        return allocations >= taskParams.getNumberOfAllocations();
+        return sumOfCurrentAllocationsOnAllStartingOrStartedNodes >= taskParams.getNumberOfAllocations();
```

Comment:
I think the use of `getTargetAllocations` here rather than `getCurrentAllocations` is a bug

Reply:
perhaps the TrainedModelsAssignment planner only sets getTargetAllocations to be LTE to the number of processors on a node, in which case, the node would be satisfied if the sum of targetAllocations is enough